src/nodes/QskArcNode.cpp

```diff
@@ -132,5 +155,23 @@ void QskArcNode::setArcData( const QRectF& rect, const QskArcMetrics& arcMetrics
     else
     {
         delete borderNode;
+        borderNode = nullptr;
+    }
+    
+    const auto oldChildCount = this->childCount();    
+    const auto newChildCount =
+        ( shadowNode ? 1 : 0 ) + ( fillNode ? 1 : 0 ) + ( borderNode ? 1 : 0 );
+     
+    if ( oldChildCount != newChildCount ) 
+    {
+        removeAllChildNodes();
+        
+        for ( QSGNode* node : { ( QSGNode* ) shadowNode, ( QSGNode* ) fillNode, ( QSGNode* ) borderNode })
+        {
+            if ( node != nullptr ) 
+            {
+                appendChildNode( node );
+            }
```

Comment:
@uwerat Is it too costly to reattach all children in the right order? Makes the implementation easy and extendable but if you want I can revert it to prepend, append and insertChildNodeAfter/Before ...

Reply:
Adapted to `QskBoxNode`'s pattern